### PR TITLE
[promptflow][internal] Support query spans via trace id & add hash header in list line runs

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR targets to refine the trace management APIs:

- Revert `evaluations` schema change happened in #2035 , also update the JS bundle for this change.
- Add query parameter `trace_ids` for `/v1.0/Spans/list`
- Add header `X-Line-Runs-List-Hash` in header for `/v1.0/LineRuns/list`: UX can use it to determine if the line runs have changed

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by brynn-code — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/_service/apis/line_run.py [hunk 1]
```diff
@@ -90,4 +76,16 @@ def get(self):
...
+            line_runs_id_list.append(line_run.line_run_id)
+            line_runs_dict.append(asdict(line_run))
+
+        # create a hash of the sorted list of line run ids
+        # and add it to the response header
+        # so that UX can use it to determine if the line runs have changed
+        line_runs_id_list.sort()
+        line_runs_hash = hashlib.sha256(",".join(line_runs_id_list).encode()).hexdigest()
+
+        return line_runs_dict, 200, {HASH_HEADER_IN_LINE_RUNS_LIST: line_runs_hash}
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/apis/line_run.py:91]: why do we need to do this in server side for UX?

Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_service/apis/line_run.py:91]: it's not a very heavy logic for server side, so implement here; I might need to sync with Kaiyi on this.

Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_service/apis/line_run.py:91]: also, we have full knowledge on whether a hash should change - one advantage from our side.

Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_service/apis/line_run.py]: synced with Kaiyi, will remove this and client will do the hash calculation

**Review by wangchao1230 — DISMISSED** [review]

**Review by brynn-code — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**